### ISSUE TITLE
[Accessibility Docs] Icons

### ISSUE THREAD

**Issue by sowasred2012** [canonical-web-and-design/vanilla-framework#4045]
Using [this template](https://docs.google.com/document/d/1IwFMfJrTfJPXDcVJfLWXYkw-pjgWhBWo4GX8X098Q64/edit#), create accessibility docs for [Icons](https://vanillframework.io/docs/patterns/icons)

**Comment by bethcollins92** [comment]
[Accessibility doc](https://docs.google.com/document/d/1B3JqDWU25pGuVdmETeM5DCxQog-9aQSciF4_wWPDAaY/edit#) for icons

**Comment by davegoddard42** [comment]
LGTM @bethcollins92 

Did you come across anything about icon colour contrast, in the same way we look at contrast ratios for text on a background? I'd guess we should use similar rules (WCAG). There's some stuff [here](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) (ctrl+f search "icons").

[This page](https://www.w3.org/WAI/WCAG21/Techniques/general/G207) is specific to icons and suggests a ratio of 3:1.

**Comment by bethcollins92** [comment]
Thanks @davegoddard42, I hadn't seen that so thanks for pointing it out. I added another point in the considerations mentioning this

**Comment by davegoddard42** [comment]
@bethcollins92 thanks - awesome job again

**Comment by petermakowski** [comment]
This looks great @bethcollins92 

Just a small suggestion - I'd include the most point the considerations section in the summary (a lot of people read _only_ the summary). Something like:

~~text can be added within the element to aid accessibility~~
text should be added unless the icon is purely decorative
